SCF-Session-9-Assignment/src/question/ShapeFactory.java

```diff
@@ -0,0 +1,114 @@
+package question;
+
+/**
+ * Class ShapeFactory is a class which determines which type of shape is to be created.
+ * The class provides shape id to each shape created.
+ * It also validates if a shape can be created for the given screen.
+ * 
+ * @author Jugal
+ * Dated 07/24/2019
+ */
+public class ShapeFactory {
+	
+	static int shapeId = 0;
+	
+	/**
+	 * Method to create a shape according to the given name and dimensions.
+	 * 
+	 * @param an array of values of origin coordinate, length, width, side, radius, etc. of the respective shape.
+	 * @return object of the respective shape.
+	 */
+	static Shape buildShapeObject(String inputShape, double[] param)
+	{
+		Shape s = null;
+		String inputShape2 = inputShape.toUpperCase();				//Converting the shape type to uppercase for comparison with enum.
+		Shape.ShapeType type = Shape.ShapeType.valueOf(inputShape2);				//Converting the input shape type to enum.
+		
+		//Checking whether the input shape type is which one. Assertion Error if invalid type is as input.
+		switch(type)
+		{
+		case RECTANGLE:
+			assert !(validateShape( param[0], param[1], param[2], param[3])) : "Rectangle cannot be created";				//Checking if the shape can be created in the screen with given dimensions.
+			
+			 s = new Rectangle( param[0], param[1], param[2], param[3], ++shapeId);
+			 break;
+			 
+		case TRIANGLE:
+			assert !(validateShape( param[0], param[1], param[2], param[3])) : "Triangle cannot be created";
+			
+			 s = new Triangle(param[0], param[1], param[2], param[3], ++shapeId); 
+			 break; 
+			 
+		case SQUARE:
+			assert !(validateSquare( param[0], param[1], param[2])) : "Square cannot be created";
+			
+			s = new Square(param[0], param[1], param[2], ++shapeId);
+			break;
+			
+		case CIRCLE:
+			//assert !(validateCircle(param)) : "Circle cannot be created";
+			if(!(validateCircle(param)))
+				throw new AssertionError();
+			
+			s = new Circle(param[0], param[1], param[2], ++shapeId);
+			break;
+			
+		default:
+			throw new AssertionError();

```

Comment:
No need to throw any error. Either show some message to the user and break or continue the entering choices.

Reply:
Done